core/src/main/scala/chisel3/Printable.scala

```diff
@@ -61,59 +61,77 @@ object Printable {
 
   /** Pack standard printf fmt, args* style into Printable
     */
-  def pack(fmt: String, data: Data*): Printable = {
+  def pack(fmt : String, data : Data*): Printable = {
     val args = data.toIterator
 
+    // The string received as an input to pack is already
+    // treated  i.e. escape sequences are processed. 
+    // Since StringContext API assumes the parts are un-treated
+    // treatEscapes is called within the implemented custom interpolators. 
+    // The literal \ needs to be escaped before sending to the custom cf interpolator. 
+    val fmtEscapeBackSlash = fmt.replace("\\","\\\\")
```

Comment:
@jackkoenig  need your inputs - this is the reason why the test within PrintfSpec was failing - basically Scala strings by default are "escape treated" unless using raw interpolator. 

But the parts Seq which StringContext API provides for interpolators expects escape sequences not treated. For most cases this is not an issue (like for \n,\t etc) but for literal \ it is an issue. Hence need to add back in the escape for the literal \. 

 At least that's what I understood so far - feel free to correct me.